src/roles/index.ts

```diff
@@ -10,37 +10,29 @@ import {
   PermissionsInput,
   Role,
   RolesPermission,
-  User,
 } from './types.js';
 import { Map } from './util.js';
 
 export interface Roles {
   listAll: () => Promise<Record<string, Role>>;
-  ofCurrentUser: () => Promise<Record<string, Role>>;
   byName: (roleName: string) => Promise<Role | null>;
-  byUser: (user: string) => Promise<Record<string, Role>>;
-  assignedUsers: (roleName: string) => Promise<Record<string, User>>;
+  assignedUserIds: (roleName: string) => Promise<string[]>;
```

Comment:
nit: I know this follows the naming in the Python client, but it feels slightly over-specified. Personally, I would expect the fact that "the strings returned by this method are user IDs" to be mentioned in the method doc, rather than be a part of the name.

Reply:
I think we can do both! Also, good reminder that these methods need docstrings :D